docs/docs/schema-customization.md

```diff
@@ -209,6 +209,51 @@ Note that you don't need to explicitly provide the Node interface fields (`id`,
 > [specifying nullability](https://graphql.org/learn/schema/#lists-and-non-null)
 > in GraphQL, i.e. if a field value is allowed to be `null` or not.
 
+#### Defining media types
+
+You can specify the media types handled by a node type using the `@mimeTypes` extension:
+
+```graphql
+type Markdown implements Node
+  @mimeTypes(types: ["text/markdown", "text/x-markdown"]) {
+  id: ID!
+}
+```
+
+The types passed in are used to determine child relations of the node.
+
+#### Defining child relations
+
+The `@childOf` extension can be used to explicitly define what node types or media types a node is a child of and immediately add `child[MyType]` or `children[MyType]` as a field on the parent.
+
+The `types` argument takes an array of strings and determines what node types the node is a child of:
+
+```graphql
+# Adds `childMdx` as a field of `File` and `Markdown` nodes
+type Mdx implements Node @childOf(types: ["File", "Markdown"]) {
+  id: ID!
+}
+```
+
+The `mimeTypes` argument takes an array of strings and determines what media types the node is a child of:
+
+```graphql
+# Adds `childMdx` as a child of any node type with the `@mimeTypes` set to "text/markdown" or "text/x-markdown"
+type Mdx implements Node
+  @childOf(mimeTypes: ["text/markdown", "text/x-markdown"]) {
+  id: ID!
+}
+```
+
+If `many: true` is set, then instead of creating a single child field on the parent, it will create multiple:
```

Comment:
I assume this affects the name (`child` vs `children`) and the type (object vs list) of the field? Might be good to mention.